container.go

```diff
@@ -233,12 +233,12 @@ func (c *container) NewTask(ctx context.Context, ioCreate cio.Creator, opts ...N
 			return nil, errors.Wrapf(errdefs.ErrInvalidArgument, "unable to resolve rootfs mounts without snapshotter on container")
 		}
 
-		// get the rootfs from the snapshotter and add it to the request
-		s, err := c.client.getSnapshotter(ctx, r.Snapshotter)
+		// get the rootfs from the mount provider and add it to the request
+		p, err := c.client.getMountProvider(ctx, r.Snapshotter)
```

Comment:
Would it make sense to add a new `MountProviderKey` so that users can specify a simple mount provider different than the snapshotter?

Reply:
Could you explain a little more for the usecase of this?  Do you want a mount provider with the same name as a snapshotter?